packages/vitest/src/node/core.ts

```diff
@@ -204,42 +205,38 @@ export class Vitest {
   }
 
   public provide<T extends keyof ProvidedContext & string>(key: T, value: ProvidedContext[T]) {
-    this.getCoreWorkspaceProject().provide(key, value)
-  }
-
-  /**
-   * @deprecated internal, use `_createCoreProject` instead
-   */
-  createCoreProject() {
-    return this._createCoreProject()
+    this.getRootTestProject().provide(key, value)
   }
 
   /**
    * @internal
    */
-  async _createCoreProject() {
-    this.coreWorkspaceProject = await WorkspaceProject.createCoreProject(this)
+  _createCoreProject() {
+    this.coreWorkspaceProject = TestProject._createBasicProject(this)
     return this.coreWorkspaceProject
   }
 
-  public getCoreWorkspaceProject(): WorkspaceProject {
+  public getRootTestProject(): TestProject {
+    if (!this.coreWorkspaceProject) {
+      throw new Error(`Root project is not initialized. This means that the Vite server was not established yet and the the workspace config is not resolved.`)
+    }
     return this.coreWorkspaceProject
   }
 
   /**
    * @deprecated use Reported Task API instead
    */
-  public getProjectByTaskId(taskId: string): WorkspaceProject {
+  public getProjectByTaskId(taskId: string): TestProject {
     const task = this.state.idMap.get(taskId)
     const projectName = (task as File).projectName || task?.file?.projectName || ''
-    return this.projects.find(p => p.getName() === projectName)
-      || this.getCoreWorkspaceProject()
+    return this.projects.find(p => p.name === projectName)
+      || this.getRootTestProject()
       || this.projects[0]
   }
 
   public getProjectByName(name: string = '') {
-    return this.projects.find(p => p.getName() === name)
-      || this.getCoreWorkspaceProject()
+    return this.projects.find(p => p.name === name)
+      || this.getRootTestProject()
```

Comment:
This public method can now throw so we should use direct access instead:

```suggestion
      || this.coreWorkspaceProject
```